web/cobrands/fixmystreet/fixmystreet.js

```diff
@@ -493,14 +666,335 @@ $.fn.drawer = function(id, ajax) {
+            '<a href="#ok" id="mob_ok">' +
+                translation_strings.ok +
+            '</a>' +
+            '</p>')
+        .css({
+            position: 'relative', // Stop map being absolute, so reporting form doesn't get hidden
+            width: width,
+            height: height
+        });
+
+        $('.mobile-map-banner').html('<a href="/">' + translation_strings.home + '</a> ' + translation_strings.right_place);
+
+        // mobile user clicks 'ok' on map
+        $('#mob_ok').toggle(function(){
+            //scroll the height of the map box instead of the offset
+            //of the #side-form or whatever as we will probably want
+            //to do this on other pages where #side-form might not be
+            $('html, body').animate({ scrollTop: height-60 }, 1000, function(){
+                $('#mob_sub_map_links').addClass('map_complete');
+                $('#mob_ok').text(translation_strings.map);
+            });
+        }, function(){
+            $('html, body').animate({ scrollTop: 0 }, 1000, function(){
+                $('#mob_sub_map_links').removeClass('map_complete');
+                $('#mob_ok').text(translation_strings.ok);
+            });
+        });
+    }
+
+    fixmystreet.page = 'new';
+  },
+
+  report: function(reportPageUrl, reportId, callback) {
+    $.ajax(reportPageUrl).done(function(html, textStatus, jqXHR) {
+        var $reportPage = $(html);
+        var $sideReport = $reportPage.find('#side-report');
+
+        if ($sideReport.length) {
+            $('#side').hide(); // Hide the list of reports
+            $('#side-report').remove(); // Remove any existing report page content from sidebar
+            $sideReport.appendTo('#map_sidebar'); // Insert this report's content
+            $('#map_sidebar').scrollTop(0);
+
+            var found = html.match(/<title>([\s\S]*?)<\/title>/);
+            var page_title = found[1];
+            document.title = page_title;
+            fixmystreet.page = 'report';
+
+            fixmystreet.mobile_reporting.remove_ui();
+            if ($('html').hasClass('mobile') && fixmystreet.map.updateSize) {
+                fixmystreet.map.updateSize();
+            }
+
+            // If this is the first individual report we've loaded, remove the
+            // "all reports" sub_map_links but store them in a global variable
+            // so we can reinsert them when the user returns to the all reports
+            // view. With #sub_map_links detached from the DOM, we set up the
+            // individual report's sub_map_links using map_controls().
+            if (!('originalSubMapLinks' in window)) {
+                window.originalSubMapLinks = $('#sub_map_links').detach();
+            }
+            fixmystreet.set_up.map_controls();
+
+            $sideReport.find('#key-tool-problems-nearby').addClass('.js-back-to-report-list');
+            fixmystreet.set_up.map_sidebar_key_tools();
+
+            fixmystreet.set_up.fancybox_images();
```

Comment:
Ah yes, guess so. Theoretically could only pull it in at the point they click on a report, but probably not worth that.